src/librustc_codegen_llvm/back/lto.rs

```diff
@@ -111,21 +110,35 @@ fn prepare_lto(
             }
 
             let archive = ArchiveRO::open(&path).expect("wanted an rlib");
-            let bytecodes = archive
+            let object_files = archive
                 .iter()
                 .filter_map(|child| child.ok().and_then(|c| c.name().map(|name| (name, c))))
-                .filter(|&(name, _)| name.ends_with(RLIB_BYTECODE_EXTENSION));
-            for (name, data) in bytecodes {
+                .filter(|&(name, _)| name.ends_with(".o"));
```

Comment:
FWIW rlibs can have non-rust `*.o` files in them if staticlibs are linked into them, but I suspect bitcode extraction will fail to find bitcode in that case anyway.

Reply:
You are correct, and I think this is why the LTO tests are still failing with this version of the code. For example, here is a sample of the object files (the `name`) that get searched for embedded bitcode in for `test/ui/debuginfo-lto.rs`:
```
std-41707e39573ec24c.10qlq9isnkfvonab.rcgu.o
panic_unwind-fb54cae34407c225.16je7xxussvyp7dt.rcgu.o
hashbrown-fdaf2869972389f3.hashbrown.4xv4kx6v-cgu.0.rcgu.o
backtrace-e4bfa7af20dc2d47.backtrace.16m3lrxn-cgu.0.rcgu.o
alloc.o
```
The first four are fine but there is a failure on the last one because it lacks bitcode, and it clearly looks different. The current code gives a fatal error if bitcode isn't found, perhaps I just need to be more forgiving and just ignore ones lacking bitcode.

Reply:
This is what I was mentioning earlier actually about "we don't actually know what files are rust object files" right now. Ideally we would have static information in rustc (probably in the metadata) for what object files are Rust object files and which are from other libraries. 

I suspect, though, that if you simply skipped object files that didn't have ".rcgu." or ones that didn't have bitcode it would be fine. We could always leave this as a TODO for fixing up later.

Reply:
Aha, we already have a `looks_like_rust_object_file` function. Maybe I'll use that, and reinsert the fatal error if bitcode extraction fails.